BlueMapCore/src/main/java/de/bluecolored/bluemap/core/world/block/entity/BlockEntity.java

```diff
@@ -0,0 +1,111 @@
+package de.bluecolored.bluemap.core.world.block.entity;
+
+import com.google.gson.reflect.TypeToken;
+import de.bluecolored.bluemap.core.logger.Logger;
+import de.bluecolored.bluenbt.BlueNBT;
+import de.bluecolored.bluenbt.NBTDeserializer;
+import de.bluecolored.bluenbt.NBTReader;
+import de.bluecolored.bluenbt.TypeDeserializer;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+@NBTDeserializer(BlockEntity.BlockEntityDeserializer.class)
+public class BlockEntity {
+    private static final BlueNBT BLUENBT = new BlueNBT();
+    private static final Map<String, Class<? extends BlockEntity>> ID_MAPPING = Map.of(
```

Comment:
How about switching this for a `Map<String, Function<Map<String, Object>, ? extends BlockEntity>>` instead?
Then we can store the type's constructor in the map directly and don't need to use reflection below :)

Reply:
Right, apart from the ungodly type we can just store a method reference to the constructor

Reply:
Can always write a small functional interface to make the type more readable :D

```java
Map<String, BlockEntityConstructor> ID_MAPPING = Map.of(
     "minecraft:sign", SignBlockEntity::new,
     "minecraft:skull", SkullBlockEntity::new,
     "minecraft:banner", BannerBlockEntity::new
);

@FunctionalInterface
private interface BlockEntityConstructor {
    BlockEntity construct(Map<String, Object> data);
}
```
super fancy :)